dao/application_dao.go

```diff
@@ -124,3 +125,21 @@ func (a *ApplicationDaoImpl) ToEventJSON(resource util.Resource) ([]byte, error)
 
 	return data, err
 }
+
+func (a *ApplicationDaoImpl) PauseAll(sourceId int64) error {
+	result := DB.Debug().
+		Model(&m.Application{}).
+		Where("source_id = ?", sourceId).
```

Comment:
Probably want to tack a `.Where("tenant_id = ?", a.TenantID)` just to be safe.

Reply:
Yeah, good idea!